pkg/webui/console/components/gateway-data-form/index.js

```diff
@@ -133,6 +143,19 @@ class GatewayDataForm extends React.Component {
 
     this.state = {
       shouldDisplayWarning: this.isNotValidDuration(props.initialValues.schedule_anytime_delay),
+      showFrequencyPlanWarning: this.isNoFrequencyPlan(props.initialValues.freqPlan),
+    }
+  }
+
+  componentDidMount() {
+    const { update } = this.props
+    const { values } = this.props.formRef.current
+
+    if (this.isNoFrequencyPlan(values.frequency_plan_id) && update) {
```

Comment:
You already have all values in `initialValues`. No need to access internal form values.

```suggestion
 	const { update, initialValues } = this.props
    if (this.isEmptyFrequencyPlan(initialValues.frequency_plan_id) && update) {
      this.setState({
        showFrequencyPlanWarning: true,
      })
    }
```

Do we actually need this? Isnt it enough to check for empty `frequency_plan_id` in `construtor`?